keps/sig-cli/20190318-kustomize-generators-transformers.md

```diff
@@ -0,0 +1,435 @@
+[Tools for generating][] a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Release Signoff Checklist](#release-signoff-checklist)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [User Stories [optional]](#user-stories-optional)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+    - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+    - [Version Skew Strategy](#version-skew-strategy)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks [optional]](#drawbacks-optional)
+  - [Alternatives [optional]](#alternatives-optional)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Enable users to extend Kustomize *Transformers* and *Generators* through `exec` plugins.  Kustomize
+provides Resource Config to plugins through STDIN, and plugins emit Resource Config back to Kustomize
+through STDOUT.
+
+- Add 2 fields to `kustomization.yaml`: `generators` and `transformers`
+- Plugins to actuate new fields have the path `$XDG_CONFIG_HOME/kustomize/plugins/<API group>`
+
+Much of the existing Kustomize functionality could be implemented through this plugin mechanism.
+
+Example Built-In Generators:
+
+- `resource`
+- `bases`
+- `secretGenerators`
+- `configMapGenerators`
+
+Example Built-In Transforms:
+
+- `commonAnnotations`
+- `commonLabels`
+- `images`
+- `namePrefix` # not supported in iteration 1 due to transformation restrictions
```

Comment:
`namePrefix` is supported.

Reply:
no it isn't, it changes the name of the resource - e.g. adds and deletes.